### PR TITLE
Issue#469: Remove PHPMailer from autoloaded files and composer files …

### PR DESCRIPTION
**Pull request by Anupamatd**
…from compiled extlib/simplesamlphp

### CODE REVIEW

**Comment by danmarsden** [comment]
is it worth adding a unit test or some other form of test to prevent us hitting this again next time we do the update? - especially as we've hit it twice so far?

**Comment by dmitriim** [comment]
yeah. Good point @danmarsden A quick test for the file presence as well as  scanning for PHPMailer in auto-lading files would help the next time I guess,

**Comment by Anupamatd** [comment]
Sure @danmarsden @dmitriim - will add that. Thank you

**Comment by Anupamatd** [comment]
Hi @dmitriim @danmarsden - added PHP unit test to cover that. Please review. Thank you

File: tests/phpmailer_test.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+defined('MOODLE_INTERNAL') || die();
+
+/**
+ * auth_saml2 PHPMailer unit tests
+ *
+ * @package    auth_saml2
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class auth_saml2_phpmailer_testcase extends advanced_testcase {
```
Comment (dmitriim) [inline tests/phpmailer_test.php:32]: In out particular case we are trying to check PHPMailer, however, I'm thinking if potentially we can add more checks that can be run after we simplesamlphp upgrade library without creating a test file for each of the case. 

Do you think it makes sense to rename this test file to something a bit more generic? E.g. auth_saml2_simplesamlphp_upgrade_testcase or something...

Comment (Anupamatd) [reply tests/phpmailer_test.php]: Thank you @dmitriim for suggesting that !! Renamed file and class name. Please review. Thank you..

**Pull request merged**